### PR TITLE
feat: implement moving entry to trash bin

### LINKED ISSUE #319: Move entry to bin

### PR DESCRIPTION
**Pull request by StarryFolf**

### CODE REVIEW

File: src/Api/Controllers/BinController.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        _currentUserService = currentUserService;
+    }
+    
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="request"></param>
+    /// <returns></returns>
+    [RequiresRole(IdentityData.Roles.Staff, IdentityData.Roles.Employee)]
```
Comment (ChienNQuang) [inline src/Api/Controllers/BinController.cs]: what is staff doing here?

File: src/Application/Digital/Commands/MoveEntryToBin.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            if (entry is null)
+            {
+                throw new KeyNotFoundException("Entry does not exist");
+            }
+            
+            var path = entry.Path;
+            var firstSlashIndex = path.IndexOf("/", StringComparison.Ordinal);
+            var binCheck = path.Substring(0, firstSlashIndex);
+
+            if (binCheck.Contains(BinString))
```
Comment (ChienNQuang) [inline src/Application/Digital/Commands/MoveEntryToBin.cs:49]: you did not account for the fact that one entry contains many other entries

**Review by ChienNQuang — CHANGES_REQUESTED** [review]

File: src/Application/Digital/Commands/MoveEntryToBin.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            var entryPath = entry.Path;
+            var firstSlashIndex = entryPath.IndexOf("/", StringComparison.Ordinal);
+            var binCheck = entryPath.Substring(0, firstSlashIndex);
+
+            if (binCheck.Contains(BinString))
+            {
+                throw new ConflictException("Entry is already in bin.");
+            }
+
+            if (!entry.Owner.Username.Equals(request.CurrentUser.Username))
```
Comment (kaitoz11) [inline src/Application/Digital/Commands/MoveEntryToBin.cs]: @StarryFolf 
Why using `username` instead of `id`?

**Review by kaitoz11 — APPROVED** [review]
LGTM

**Review by ChienNQuang — APPROVED** [review]
LGTM

**Pull request merged**